From: Goldwyn Rodrigues <rgoldwyn@suse.com>
Subject: Correct calls to to_ib_ah_attr()
Patch-mainline: no
References: bnc#806847

The change was introduced by 4c3eb3ca13966508bcb64f39dcdef48be22f1731
where prototype of to_ib_ah_attr() was changed, however all function
calls were not converted.

Also fixed other compiler warnings wrt __devinit and vlan initialization.

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@suse.com>
--- 
Index: linux-3.0-SLES11-SP2/drivers/infiniband/hw/mlx4/qp.c
===================================================================
--- linux-3.0-SLES11-SP2.orig/drivers/infiniband/hw/mlx4/qp.c	2013-03-01 12:11:23.000000000 -0600
+++ linux-3.0-SLES11-SP2/drivers/infiniband/hw/mlx4/qp.c	2013-03-01 15:46:11.000000000 -0600
@@ -1339,7 +1339,7 @@ static int build_mlx_header(struct mlx4_
 	int is_eth;
 	int is_vlan = 0;
 	int is_grh;
-	u16 vlan;
+	u16 vlan = 0;
 
 	send_size = 0;
 	for (i = 0; i < wr->num_sge; ++i)
@@ -2307,8 +2307,8 @@ int mlx4_ib_query_xrc_rcv_qp(struct ib_x
 
 	if (qp->ibqp.qp_type == IB_QPT_RC || qp->ibqp.qp_type == IB_QPT_UC ||
 	    qp->ibqp.qp_type == IB_QPT_XRC) {
-		to_ib_ah_attr(dev->dev, &qp_attr->ah_attr, &context.pri_path);
-		to_ib_ah_attr(dev->dev, &qp_attr->alt_ah_attr,
+		to_ib_ah_attr(dev, &qp_attr->ah_attr, &context.pri_path);
+		to_ib_ah_attr(dev, &qp_attr->alt_ah_attr,
 			      &context.alt_path);
 		qp_attr->alt_pkey_index = context.alt_path.pkey_index & 0x7f;
 		qp_attr->alt_port_num	= qp_attr->alt_ah_attr.port_num;
Index: linux-3.0-SLES11-SP2/drivers/net/mlx4/xrcd.c
===================================================================
--- linux-3.0-SLES11-SP2.orig/drivers/net/mlx4/xrcd.c	2013-03-01 14:35:07.000000000 -0600
+++ linux-3.0-SLES11-SP2/drivers/net/mlx4/xrcd.c	2013-03-01 15:47:33.000000000 -0600
@@ -54,7 +54,7 @@ void mlx4_xrcd_free(struct mlx4_dev *dev
 }
 EXPORT_SYMBOL_GPL(mlx4_xrcd_free);
 
-int __devinit mlx4_init_xrcd_table(struct mlx4_dev *dev)
+int mlx4_init_xrcd_table(struct mlx4_dev *dev)
 {
 	struct mlx4_priv *priv = mlx4_priv(dev);
 
